### PR TITLE
Fix cross appdomain exception

### PR DESCRIPTION
**Pull request by BrennanConroy**
Related: https://github.com/aspnet/Caching/issues/203

### CODE REVIEW

**Comment by BrennanConroy** [comment]
@davidfowl @pranavkm

File: test/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Tests/DataStoreErrorLoggerTest.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        private static DataStoreErrorLogger SetUp(Func<string, LogLevel, bool> filter = null, string name = null)
+        {
+            // Arrange
+            var provider = new DataStoreErrorLoggerProvider();
+            var logger = (DataStoreErrorLogger)provider.CreateLogger(name ?? _name);
+
+            return logger;
+        }
+
+#if NET451
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Tests/DataStoreErrorLoggerTest.cs]: Could this type be `#if NET451`?

**Comment by pranavkm** [comment]
:shipit:

File: test/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Tests/DataStoreErrorLoggerTest.cs (hunk 1 above)
Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Tests/DataStoreErrorLoggerTest.cs]: What?

Comment (pranavkm) [inline test/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Tests/DataStoreErrorLoggerTest.cs]: Could we move the `#ifdef` to the top of the class?

Comment (BrennanConroy) [inline test/Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.Tests/DataStoreErrorLoggerTest.cs]: Oh yeah, that's fine

**Comment by BrennanConroy** [comment]
🆙 📅 
Looks like there was an issue with not using ObjectHandle

**Comment by pranavkm** [comment]
:shipit: again.

**Pull request merged**